nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/test/java/org/apache/nifi/dbcp/DBCPServiceTest.java

```diff
@@ -23,48 +23,49 @@
 import org.apache.nifi.reporting.InitializationException;
 import org.apache.nifi.util.TestRunner;
 import org.apache.nifi.util.TestRunners;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.io.TempDir;
 
 import java.io.File;
+import java.nio.file.Path;
 import java.sql.Connection;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.ArrayList;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
+@DisabledOnOs(OS.WINDOWS)
```

Comment:
What was the reason for disabling this test on Windows?

Reply:
I am pretty sure the reason I disabled it was that when I ran it as-is, the JUnit temporary folder capabilities ran into a file locking issue on the Windows build. Pretty sure I've seen that behavior in the past with the Windows build. Been a while since I ran it, so I think that's what it was.

Reply:
Thanks for the reply @MikeThomsen, I looked at the previous automated build and observed the errors you mentioned:

```
java.io.IOException: Failed to delete temp directory C:\Users\RUNNER~1\AppData\Local\Temp\junit8996089088338771159. The following paths could not be deleted (see suppressed exceptions for details): , target, target\db, target\db\db.lck
```

That does raise a question as to why the temporary directory appears to be using the `target` directory.  Rather than disabling the test on Windows, it looks like the directory should be changed to use an actual temporary directory, instead of something under the `target` directory.  I would expect that using a regular temporary directory would allow it to work, since the test was previously working on JUnit 4.  Can you take a look at making that adjustment?

Reply:
I tried out a few things, and it looks like there might not be a clean way to tear this down on Windows.

Reply:
Thanks for trying out some options @MikeThomsen. It appears that Derby still has an open handle to the database files for some reason.  Did you consider creating and removing the directory manually in `BeforeAll` and `AfterAll` methods?  Given that this is a popular component, it seems like a step backwards to disable it as part of the JUnit 5 changes.

Reply:
As one other option for now, I would be fine with leaving the DBCPServiceTest as it is with JUnit 4 and creating a new Jira issue to revisit it later. That could allow the rest of this PR to move forward.

Reply:
I think we can skip this on Windows because it's a problem with Windows and not the Unit tests. They run just fine on macOS and Linux.

Reply:
Thanks @MikeThomsen, I'm inclined to move ahead with merging given all of the other improvements in this PR.

I looked at `DatabaseRecordSinkTest` and noticed that it uses the Derby DB and similar structure, but it runs on Windows. It seems like this could be related to the fact that it uses a custom `AfterAll` method that ignores `IOExceptions` when deleting the database location, however, that also allows it run on all platforms. Although this is minor, it avoids one step backward. Can you take one more look at using the cleanup approach from `DatabaseRecordSinkTest`? If that doesn't work, then I'm fine with moving forward disabling on Windows, but since it is so similar, it seems worth one more try.

Reply:
Ok. I'll compare the two tests and see if there's something off about DBCPServiceTest.

Reply:
Thanks. I'll rebase right now.